tools/umi_tools/umi_tools.yml

```diff
@@ -0,0 +1,45 @@
+name: umi_tools
+version: 1.0
+description: Tools for dealing with Unique Molecular Identifiers (UMIs)/Random Molecular Tags (RMTs) and single cell RNA-Seq cell barcodes. 
+keywords:
+    - UMI
+    - RMT
+    - Barcode
+tools:
+    - umi_tools:
+        description: |
+            Tools for dealing with Unique Molecular Identifiers (UMIs)/Random Molecular Tags (RMTs) and single cell RNA-Seq cell barcodes. 
+        homepage: https://github.com/CGATOxford/UMI-tools
+        documentation: https://umi-tools.readthedocs.io/en/latest/
+processes:
+    - dedup:
+        operation: |
+            Set command args to params.umitools_dedup_args
+            The program will execute with the following pattern:
+                umi_tools dedup --log={SAMPLE_ID}.dedup.log {params.umitools_dedup_args} -I {SAMPLE_ID}.bam -S {SAMPLE_ID}.dedup.bam --output-stats={SAMPLE_ID}
+        description: |
+            Groups PCR duplicates and de-duplicates reads to yield one read per group. 
+            Use this when you want to remove the PCR duplicates prior to any downstream analysis.
+        input:
+            - sample_id:
+                type: string
+                description: Sample identifier
+            - bai:
+                type: file
+                description: BAM file index
+            - bam:
+                type: file
+                description: BAM sequence file
+        output:
+            - dedupBam:
```

Comment:
Assuming `samtools` isnt installed as a dependency? If it is would be good to output the index here too.

Reply:
As we are building everything in this PR should be no problem to add it..

Reply:
yes agreed